memstore/archive_store.go

```diff
@@ -374,6 +374,7 @@ func (b *ArchiveBatch) BuildIndex(sortColumns []int, primaryKeyColumns []int, pk
 func (b *ArchiveBatch) Clone() *ArchiveBatch {
 	newBatch := &ArchiveBatch{
 		Batch: Batch{
+			RWMutex: b.Batch.RWMutex,
```

Comment:
need to copy by reference. therefore we may need to change the lock to a pointer inside batch